File: global/src/main/java/com/ms2709/global/WebAdapter.java
package com.ms2709.global;

import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Component;

import java.lang.annotation.*;

/**
 * WebAdapter의 역할을 나타내기 위한 어노테이션
 *
 * @author 심문섭
 * @version 1.0
 * @class WebAdapter
 * @since 2024-05-02 오전 8:27
 */
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Component
public @interface WebAdapter {
    /**
     * The value may indicate a suggestion for a logical component name,
     * to be turned into a Spring bean in case of an autodetected component.
     * @return the suggested component name, if any (or empty String otherwise)
     */
    @AliasFor(annotation = Component.class)
    String value() default "";

}

File: member-service/src/main/kotlin/ms2709/member/global/WebAdapter.kt
package ms2709.member.global

import org.springframework.web.bind.annotation.RestController
import java.lang.annotation.Documented

/**
 *
 * WebAdapter의 역할을 나타내기 위한 어노테이션
 *
 * @class WebAdapter
 * @author 심문섭
 * @version 1.0
 * @since 2024-04-29 8:58 AM
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@RestController
annotation class WebAdapter(val value: String = "") {

}